robotoff/spellcheck/pipeline_spellchecker.py

```diff
@@ -0,0 +1,58 @@
+from dataclasses import asdict
+from typing import List, Dict, Optional
+
+from robotoff.spellcheck.items import SpellcheckItem
+from robotoff.spellcheck.patterns import PatternsSpellchecker
+from robotoff.spellcheck.percentages import PercentagesSpellchecker
+from robotoff.spellcheck.vocabulary import VocabularySpellchecker
+from robotoff.spellcheck.elasticsearch import ElasticSearchSpellchecker
+
+from robotoff.spellcheck.exceptions import PipelineSpellcheckerException
+
+
+class PipelineSpellchecker:
+
+    SPELLCHECKERS = {
+        "elasticsearch": ElasticSearchSpellchecker,
+        "patterns": PatternsSpellchecker,
+        "percentages": PercentagesSpellchecker,
+        "vocabulary": VocabularySpellchecker,
+    }
+
+    def __init__(self):
+        self.spellcheckers = []
+        self.item = None
```

Comment:
I'm not comfortable with PipelineSpellchecker being tied to the item. In my opinion, the PipelineSpellchecker should generate and return a SpellcheckItem. Methods on this item are then used to fetch various information about the corrections.

Reply:
Completely agree with you. I did not thought about returning an Item since I wanted an easy interface "text_to_correct -> corrected_text" but the result was complicated for nothing.
Now I implemented your suggestion : https://github.com/openfoodfacts/robotoff/pull/181/commits/cc6c5b5df1aba2b70ec611ddedfc1bcb409191f4